paramiko/file.py

```diff
@@ -372,7 +372,10 @@ def write(self, data):
 
         :param str/bytes data: data to write
         """
-        data = b(data)
+        if isinstance(data, text_type):
+            # GZ 2017-05-25: Accepting text on a binary stream unconditionally
+            # cooercing to utf-8 seems questionable, but compatibility reasons?
```

Comment:
Truly binary data would not be `text_type` (`str` for py2, `unicode` for py2), it would be bytes or something like that, so it should be passed through fine. If someone calls `write()` with a unicode type, this kicks in, turning unicode to utf-8 bytes. This seems like the best thing to do if someone passes unicode here.